services/src/main/java/org/keycloak/theme/DefaultThemeSelectorProvider.java

```diff
@@ -50,9 +48,9 @@ public String getThemeName(Theme.Type type) {
 
         if (name == null || name.isEmpty()) {
             name = Config.scope("theme").get("default", Version.NAME.toLowerCase());
-            if ((type == Theme.Type.ACCOUNT) && isAccount2Enabled) {
+            if ((type == Theme.Type.ACCOUNT) && Profile.isFeatureEnabled(Profile.Feature.ACCOUNT2)) {
                 name = name.concat(".v2");
-            } else if ((type == Theme.Type.ADMIN) && isAdmin2Enabled) {
+            } else if ((type == Theme.Type.ADMIN) && Profile.isFeatureEnabled(Profile.Feature.ADMIN2)) {
```

Comment:
This is necessary for Undertow as it shares the same JVM with tests and the static fields previously were not re-instantiated on server restart (or disabled features update).

Although updating server code to fix tests is not ideal, I believe this change actually aligns the use of `Profile.isFeatureEnabled()` with other parts of Keycloak – we usually don't store the result `Profile.isFeatureEnabled()` of in a constant.